src/parser/device.rs

```diff
@@ -17,26 +17,34 @@ impl SubParser for Matcher {
     type Item = Device;
 
     fn try_parse(&self, text: &str) -> Option<Self::Item> {
-        if let Some(captures) = self.regex.captures(text) {
+        if let Ok(Some(captures)) = self.regex.captures(text) {
             let family: String =
                 if let Some(device_replacement) = &self.device_replacement {
                     replace(&device_replacement, &captures)
                 } else {
-                    captures.at(1).map(str::to_string)?
+                    captures
+                        .get(1)
+                        .map(|x| x.as_str())
+                        .and_then(none_if_empty)
+                        .map(ToString::to_string)?
                 };
 
             let brand: Option<String> =
                 if let Some(brand_replacement) = &self.brand_replacement {
                     none_if_empty(replace(&brand_replacement, &captures))
                 } else {
-                    captures.at(2).map(str::to_string)
+                    None
                 };
 
             let model: Option<String> =
                 if let Some(model_replacement) = &self.model_replacement {
                     none_if_empty(replace(&model_replacement, &captures))
                 } else {
-                    captures.at(3).map(str::to_string)
+                    captures
+                        .get(1)
```

Comment:
This behavior was ported from the official Python implementation and is necessary for making the tests pass. I have read the "specification" document of uap-core and I have no idea what the intended behavior is or why this used to work before.